tools/replace_code_block.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python3
+
+import argparse
+import os
+import sys
+from traceback import print_tb
+from typing import AnyStr
+
+status = 0
+
+
+def check_for_code(file: AnyStr, check: bool):
+    """Check a file and replace ``.. code::`` to ``.. code-block::``
+
+    :param file: The file to open
+    :param check: Whether or not the program is operating in ``check`` mode.
+        In check mode, the program will exit with a status 1 if files will be
+        replaced. In non-check mode, the program will replace files.
+    """
+    global status
+    with open(file, "rb") as infile:
+        content = infile.read()
+    old_content = content
+    new_content = content.replace(b".. code::", b".. code-block::")
+    if old_content != new_content:
```

Comment:
To avoid subsequent code nesting, can we flip this around to `return 0` when `old_content == new_content`?

Reply:
That can be managed.